src/lib/mongoose-helper/filter.js

```diff
@@ -0,0 +1,74 @@
+import { camelizeKeys } from "humps";
+import { isNil } from "lodash";
+
+// 目前用空格作为分隔符，可以增加新的分隔符
+const quote = (str = "") =>
+  str
+    .split(" ")
+    .map(frag => `"${frag}"`)
+    .join(" ");
+
+/**
+ * Build filter for list function
+ *
+ * @constructor
+ * @param {Object} raw raw filter
+ * @param {Schema} schema mongoose schema
+ * @returns {Object} normalized filter
+ */
+export default function build(raw, schema) {
+  const filter = Object.keys(raw).reduce((acc, key) => {
+    // TODO: 大部分逻辑移到 qn 里了，后面mongoose helper 需要从 qn 的标准query object 到 mongoose query 做转变
```

Comment:
hmm, it this code snippet written by you or you borrowed it from somewhere?

Reply:
I borrowed `mongoose-helper` from @36node/mongoose-helper, and made some tiny modification to meet my design setting. It's very convient to use this to design `list` API for me.